pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -2689,3 +2689,31 @@ ALTER TABLE t_89025 ADD COLUMN j INT DEFAULT (crdb_internal.lease_holder('a'));
 
 statement ok
 DROP TABLE t_89025
+
+# Tests for issue #90570 where we did not properly honour mutations on virtual
+# computed columns, which could cause selects to fail. Specifically in a cascaded
+# drop scenario, where the dependent column for a computed one is no longer
+# available.
+subtest drop_column_cascade_with_access
+
+statement ok
+create table t_with_dropped_index_expr (i INT PRIMARY KEY, j INT, INDEX((j+1)));
+
+statement ok
+SET CLUSTER SETTING jobs.debug.pausepoints="newschemachanger.before.exec";
+
+
+skipif config local-legacy-schema-changer
+statement error job \d+ was paused before it completed with reason: pause point "newschemachanger.before.exec" hit
+ALTER TABLE t_with_dropped_index_expr DROP COLUMN j CASCADE;
+
+statement ok
+SET CLUSTER SETTING jobs.debug.pausepoints="";
```

Comment:
why do we want to clear the pause point? don't we want the column stay as `DELETE_ONLY` so we can make sure the following `SELECT` won't fail on the virtual column?

Reply:
@chengxiong-ruan Once the pre-commit phases is done and the first job is kicked off we should be in DELETE ONLY, clearing the pause point won't resume the job. So, it will just stay in that state.